src/facade/parsed_command.cc

```diff
@@ -0,0 +1,167 @@
+// Copyright 2025, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#include "facade/parsed_command.h"
+
+#include "base/logging.h"
+#include "facade/conn_context.h"
+#include "facade/dragonfly_connection.h"
+#include "facade/reply_builder.h"
+
+namespace facade {
+
+using namespace std;
+
+namespace {
+
+struct PayloadVisitor {
+  void operator()(monostate) {
+  }
+
+  void operator()(long v) {
+    rb->SendLong(v);
+  }
+
+  void operator()(double v) {
+    LOG(FATAL) << "Double replies are not supported in MC protocol";
+  }
+
+  void operator()(const payload::SimpleString& ss) {
+    rb->SendSimpleString(ss);
+  }
```

Comment:
You are reinventing something here, there is `CapturingReplyBuilder::Apply(Payload&& pl, RedisReplyBuilder* rb)`